sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreePatterns.scala

```diff
@@ -134,6 +134,7 @@ object TreePattern extends Enumeration  {
   val UNRESOLVED_WINDOW_EXPRESSION: Value = Value
 
   // Unresolved Plan patterns (Alphabetically ordered)
+  val TABLE_VALUED_FUNCTION_WITH_ALIAS: Value = Value
```

Comment:
Since we already have `UNRESOLVED_TABLE_VALUED_FUNCTION`, according to the convention,
`TABLE_VALUED_FUNCTION_WITH_ALIAS` -> `UNRESOLVED_TABLE_VALUED_FUNCTION_WITH_ALIAS`?

Reply:
It contains resolved TVF, but the column alias is not resolved yet.

Reply:
I was thinking about using `UNRESOLVED_TABLE_VALUED_FUNCTION_WITH_ALIAS` but we need to distinguish between 
1) an unresolved table-valued function with a string function name and 
2) an unresolved table-valued function that has been looked up in the function registry, but its function body (logical plan) and output aliases have not been resolved yet.

So if we use `UNRESOLVED_TABLE_VALUED_FUNCTION_WITH_ALIAS`, it can be confusing to make it a unary node vs a leaf node like `UNRESOLVED_TABLE_VALUED_FUNCTION`.

Let me know what you think @dongjoon-hyun